### ISSUE TITLE
`cirq_google.run_calibrations` should work with `SimulatedLocalEngine`s

### ISSUE THREAD

**Issue by augustehirth** [quantumlib/Cirq#5978]
[Floquet calibration](https://quantumai.google/cirq/noise/floquet_calibration_example) is a notebook that was designed to use the currently offline QCS to demonstrate how to calibrate a circuit to compensate for the errors that would arise by running it on a (QCS) device. 

Without supplying QCS credentials, the notebook defaults to using a https://github.com/quantumlib/Cirq/blob/8ea41a8aa8ed2faeb3271e16878ce9189ba14431/cirq-google/cirq_google/engine/simulated_local_engine.py#L31 per this line: https://github.com/quantumlib/Cirq/blob/8ea41a8aa8ed2faeb3271e16878ce9189ba14431/cirq-google/cirq_google/engine/qcs_notebook.py#L123

The notebook later fails because `SimulatedLocalProcessor`'s `run_calibration_async` function is `NotImplemented`: https://github.com/quantumlib/Cirq/blob/8ea41a8aa8ed2faeb3271e16878ce9189ba14431/cirq-google/cirq_google/engine/simulated_local_processor.py#L276-L277
Note: For now, you need to set `processor_id` to `weber` to get past the first bug in this notebook.

For reference,  `EngineProcessor` has an implementation: 
https://github.com/quantumlib/Cirq/blob/fe67fcbe58d678bf27eeda918216706ae2994cb1/cirq-google/cirq_google/engine/engine_processor.py#L175
that calls `Engine`'s implementation: 
https://github.com/quantumlib/Cirq/blob/fe67fcbe58d678bf27eeda918216706ae2994cb1/cirq-google/cirq_google/engine/engine.py#L409
that calls `EngineProgram`'s implementation: 
https://github.com/quantumlib/Cirq/blob/fe67fcbe58d678bf27eeda918216706ae2994cb1/cirq-google/cirq_google/engine/engine_program.py#L213
that actually creates the job. 

---
It would be useful to be able to run calibrations on simulated engines, for the sake of: 
1. Testing workflows with simulated engines before slotting in a hardware-backed engine.
2. Teaching how calibration works and demonstrating it's effects without access to hardware.
3. Experimenting with modified and new calibration procedures without access to hardware.

Specifically, let's implement `cirq_google.simulated_local_processor.SimulatedLocalProcessor.run_calibration_async`  to have the same effective functionality as `cirq_google.engine_processor.EngineProcessor.run_calibration_async`. It should be able to handle the same input and produce the same output, with the caveat that any circuit executions are instead performed by the simulated processor.

I'm not sure what the scope of this request is, since `SimulatedLocalProcessor` and `EngineProcessor` seem to be relatively parallel constructs, and there seems to be a decent amount of complexity in handling protobuf'd requests.

Tentative priority: P2?

**Comment by augustehirth** [comment]
@dstrain115 @tanujkhattar

**Comment by dstrain115** [comment]
This is no longer relevant after #6442 removed this functionality.
